### PR TITLE
Only compare release version and chart values when updating

### PR DESCRIPTION
**Pull request by rossf7**
Fixes a bug with updating chart values.

### CODE REVIEW

File: service/controller/v2/resource/chart/update.go [hunk 1]
```diff
@@ -95,7 +94,7 @@ func (r *Resource) newUpdateChange(ctx context.Context, obj, currentState, desir
 
 	r.logger.LogCtx(ctx, "level", "debug", "message", "finding out if the chart has to be updated")
 
-	if currentChartState.ReleaseVersion != "" && !reflect.DeepEqual(desiredChartState, currentChartState) {
```
Comment (rossf7) [inline service/controller/v2/resource/chart/update.go]: Bug meant this was always returning true. The release status is set for the current state but not the desired state.

File: service/controller/v2/resource/chart/resource.go [hunk 2]
```diff
@@ -69,6 +71,21 @@ func (r *Resource) Name() string {
...
+func isChartModified(a, b ChartState) bool {
+	// ChartValues have changed so we need to update.
+	if !reflect.DeepEqual(a.ChartValues, b.ChartValues) {
+		return true
+	}
+
+	// ReleaseVersion has changed for the channel so we need to update.
+	if a.ReleaseVersion != b.ReleaseVersion {
+		return true
+	}
```
Comment (rossf7) [inline service/controller/v2/resource/chart/resource.go]: In v1 we just reconcile on the ReleaseVersion. So this is now correct.

File: service/controller/v2/resource/chart/resource.go [hunk 3]
```diff
@@ -69,6 +71,21 @@ func (r *Resource) Name() string {
 	return Name
 }
 
+func isChartModified(a, b ChartState) bool {
+	// ChartValues have changed so we need to update.
```
Comment (fgimenez) [inline service/controller/v2/resource/chart/resource.go]: Nitpick, we need to update if the next `if` condition evaluates to true, is that correct?

File: service/controller/v2/resource/chart/update.go (hunk 1 above)
Comment (fgimenez) [reply service/controller/v2/resource/chart/update.go]: Very good catch!

File: service/controller/v2/resource/chart/resource.go [hunk 4]
```diff
@@ -69,6 +71,21 @@ func (r *Resource) Name() string {
...
+	if !reflect.DeepEqual(a.ChartValues, b.ChartValues) {
+		return true
+	}
+
+	// ReleaseVersion has changed for the channel so we need to update.
```
Comment (fgimenez) [inline service/controller/v2/resource/chart/resource.go]: Same as above, the comment assumes that the check is going to be true.

File: service/controller/v2/resource/chart/update_test.go [hunk 5]
```diff
@@ -34,7 +34,21 @@ func Test_Resource_Chart_newUpdateChange(t *testing.T) {
 			expectedUpdateState: nil,
 		},
 		{
-			description: "case 1: nonempty current state, different release version in desired state, expected desired state",
+			description: "case 1: nonempty current state, equal desired state, empty update change",
```
Comment (fgimenez) [inline service/controller/v2/resource/chart/update_test.go]: Nice to modify/extend existing tests without explicitly testing `isChartModified`, only its external effect :+1:

File: service/controller/v2/resource/chart/resource.go (hunk 3 above)
Comment (rossf7) [reply service/controller/v2/resource/chart/resource.go]: Good point. I'm going to make the ReleaseVersion check first. As if it has changed we need to update the Helm release with the new chart version. 

If the values have changed we just need to update them for the current Helm release. I'll update the comments to make them clearer.

Comment (rossf7) [reply service/controller/v2/resource/chart/resource.go]: Done.

File: service/controller/v2/resource/chart/resource.go (hunk 4 above)
Comment (rossf7) [reply service/controller/v2/resource/chart/resource.go]: Done.

**Pull request merged**